internal/bmcutils/bmcutils.go

```diff
@@ -175,3 +177,49 @@ func CreateBMCClient(
 func GetServerNameFromBMCandIndex(index int, bmc *metalv1alpha1.BMC) string {
 	return fmt.Sprintf("%s-%s-%d", bmc.Name, "system", index)
 }
+
+func ResetBMC(ctx context.Context, c client.Client, bmcObj *metalv1alpha1.BMC) error {
+	username, password, err := GetBMCCredentialsForBMCSecretName(ctx, c, bmcObj.Spec.BMCSecretRef.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get credentials from BMC secret: %w", err)
+	}
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{ssh.Password(password)},
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+			return nil
+		},
+		Timeout: 5 * time.Second,
```

Comment:
```suggestion
		Timeout: 5 * time.Minutes,
```
I just testes on one problematic BMC: it takes 50 seconds until password prompt appeared after executing `ssh user@<bmc-ip>` + another 80 seconds to log in.

Maybe we need to introduce a general SSH timeout?